backend/src/index.ts

```diff
@@ -384,6 +385,20 @@ handlers[API_PROFILE] = async (input) => {
   };
 };
 
+handlers[API_PROFILE_UPDATE] = async (input) => {
+  const params = input.url.searchParams;
+  const user = await input.requireAuthedUser();
+  user.username = params.get('username');
```

Comment:
I'd use expectStringParam here for the username and bio. Facebook uses 50 characters for username so that sounds good, and for the bio, lets say 1000.